### PR TITLE
Feat: Add /redirect command

### PR DESCRIPTION
**Pull request by NanoCode012**
## What kind of change does this PR introduce?

Adds `/redirect <user>` command following https://github.com/supabase-community/supabase-community-bot/discussions/2#discussioncomment-4415275

## What is the current behavior?

Nothing.

## What is the new behavior?

Using, `/redirect <user>`, it will reply with 
```
Hello <user>! Please use the #help-channel for questions. You will find it easier to get answers there!
```

## Additional context

~Require new `.env` variable called `HELP_CHANNEL_ID`~

image

Edit 1: Renamed `/help` to `/redirect` following discussions below.

### CODE REVIEW

**Comment by Olyno** [comment]
Hi @NanoCode012 👋🏻

Thank you for your contribution! I really like the idea. I was planning to integrate a similar command but parsing the messages with chatgpt. So if the message looks like a help request, then the bot asks the user to go to the help channel, or even create a post for him.

However, a simple command is just as effective. I'm just not sure about the term "help" for the command, in the sense that the "/help" commands are more there to display the list of commands. Something like "/redirect" would seem a bit better to me.

Regarding the message displayed, it is not dynamic/changeable, which can be inconvenient if a moderator wants to change it. Same thing for the channel id. Maybe storing *for now* the content and id in variables could make the job? In the near future, the bot will be connected to a Supabase instance, which will be easier to store this kind of information.

What do you think about it?

**Comment by NanoCode012** [comment]
Thank you for the review @Olyno .

> I'm just not sure about the term "help" for the command, in the sense that the "/help" commands are more there to display the list of commands.

I agree. The naming should be changed. I noticed this shortly after testing it later.

>  it is not dynamic/changeable, which can be inconvenient if a moderator wants to change it. Same thing for the channel id. Maybe storing for now the content and id in variables could make the job

Right now, the channel id is stored in `.env`. I didn't want to hardcode it within the script, but that's easy to change. For the content, it might be a bit more tricky as we insert the user's name and channel ID into it. Therefore, we would need placeholders in this text. It is still possible to do so. 

> I was planning to integrate a similar command but parsing the messages with chatgpt. So if the message looks like a help request, then the bot asks the user to go to the help channel, or even create a post for him.

I also thought of using AI to detect text but thought that a) simple first would be good b) I was not certain if Supabase wanted to use openai in the first place. I think this can be improved in later versions. It would have to read messages in the chat instead of slash commands however. I am not sure what's Discord plan on this feature for large server and whether bot needs to be verified.

---

In summary, I will rename the routes and move the variables to hardcoded constants instead for now. Feel free to tell me if I have made any mistakes.

**Comment by NanoCode012** [comment]
Hello @Olyno , I have updated the code as per above. Thank you.

File: src/commands/redirect.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { CommandInteraction, SlashCommandBuilder } from 'discord.js';
+
+const questionsChannelId = '1006358244786196510';
+const redirectReplyMessage =
+  `Hello {user}! Please use the {channel} channel for questions. ` +
+  'You will find it easier to get answers there!';
```
[the remark this answers is not in the document]
Comment (NanoCode012) [reply src/commands/redirect.ts:6]: Yes! I got you now. I was a bit confused when I was coding this part.

File: src/commands/redirect.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export async function execute(interaction: CommandInteraction) {
+  const userId = interaction.options.getUser('user')?.id;
+  
+  let msg = redirectReplyMessage.replace(
+    '{user}',
+    userId ? `<@${userId}>` : 'there'
+  );
+
+  msg = msg.replace('{channel}', `<#${questionsChannelId}>`);
```
[the remark this answers is not in the document]
Comment (NanoCode012) [reply src/commands/redirect.ts:27]: For the `user`, it would need to be replaced in if we want it to mention the user (like in my initial screenshot). The channel ID is not required to be a different config, it can be in the raw content.

What would you propose to be able to mention the user?

File: src/commands/redirect.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (NanoCode012) [reply src/commands/redirect.ts]: I have updated this part to a `config` obj in the `config` file instead.

**Review by Olyno — APPROVED** [review]
There is surely room for improvement, but for now it is already a good pull request. Welcome on board!
